README.md

```diff
@@ -54,6 +54,67 @@ Example configuration:
+
+1. If there's an existing field that will be used as the document ID then use the `ReplaceField` transform.  See the config example below that renames `name` to `_id` field:
+    ```
+    transforms=RenameField
+    transforms.RenameField.type=org.apache.kafka.connect.transforms.ReplaceField$Value 
+    transforms.RenameField.renames=name:_id
+    ```
+2. If you have `_id` fields and would prefer to have Cloudant generate a UUID for the document ID, use the `ReplaceField` transform to exclude the existing `_id` field:
+    ```
+    transforms=ReplaceField
+    transforms.ReplaceField.type=org.apache.kafka.connect.transforms.ReplaceField$Value 
+    transforms.ReplaceField.exclude=_id
+    ```
+3. If you have messages where the `_id` field is absent or `null` then Cloudant will generate
+a document ID. If you don't want this to happen then set an `_id` (see earlier examples).
+Alternatively filter out those documents. For example if you have messages where the `_id`
+field is `null` then you'll need to use a transform and predicate to filter out and remove this
+field:
+    ```
+    TODO 
+    ```
+
+4. If you have records where there is no value e.g. tombstones (and don't want the Cloudant sink connector to generate an empty doc with a generated ID) then 
+you'll need to use a `dropNullRecords` transform and predicate to filter out and remove these tombstone records:  
+    ```
+    transforms=dropNullRecords
+    transforms.dropNullRecords.type=org.apache.kafka.connect.transforms.Filter
+    transforms.dropNullRecords.predicate=isNullRecord
+
+    predicates=isNullRecord
+    predicates.isNullRecord.type=org.apache.kafka.connect.transforms.predicates.RecordIsTombstone
+    ```
+
+5. If you want to use the message key or another custom value as the document ID then use the `cloudant_doc_id` custom header.
+   The value set in this custom header will be added to the `_id` field.  If the `_id` field already exists then it will be overwritten
+   with the value in this header.
+   You can use the `HeaderFrom` SMT to move or copy a key to the custom header. The example config below adds the transform to move 
+   the `docid` record key to the `cloudant_doc_id` custom header and sets the header converter to string:
+   ```
+   transforms=moveFieldsToHeaders
+   transforms.moveFieldsToHeaders.type=org.apache.kafka.connect.transforms.HeaderFrom$Key
+   transforms.moveFieldsToHeaders.fields=docid
+   transforms.moveFieldsToHeaders.headers=cloudant_doc_id
+   transforms.moveFieldsToHeaders.operation=move
+   
+   header.converter=org.apache.kafka.connect.storage.StringConverter
+   ```
+
+   **Note**: The `header.converter` is required to be set to `StringConverter` since the document ID field only supports strings.
+
+**Note**: For any of the SMTs above, if the field does not exist it will skip over that message and continue processing the next message.
```

Comment:
wdyt:
```suggestion
**Note**: For any of the SMTs above, if the field does not exist it will leave the message unmodified and continue processing the next message.
```